src/fsharp/FSharp.Core/control.fs

```diff
@@ -907,9 +907,11 @@ namespace Microsoft.FSharp.Control
         let doneA           = 
             resultA()
 
+        let inline disposeIfNotNull (r:IDisposable) () = if LanguagePrimitives.IsNonNull r then r.Dispose() else ()
```

Comment:
@dsyme please tell me there is a good explanation why this works, but manually inlining this in usingA does not.